src/core/cpu/arm_interpret.cpp

```diff
@@ -886,6 +892,53 @@ uint32_t load_store_shift_reg(ARM_CPU& cpu, uint32_t instr)
     return reg;
 }
 
+void arm_sel(ARM_CPU &cpu, uint32_t instr)
+{
+    int reg1 = (instr >> 16) & 0xF;
+    int dest = (instr >> 12) & 0xF;
+    int reg2 = instr & 0xF;
+
+    uint32_t source1 = cpu.get_register(reg1);
+    uint32_t source2 = cpu.get_register(reg2);
+
+    uint32_t dest_word = 0;
+
+    const PSR_Flags* cpsr = cpu.get_CPSR();
+    for (int i = 0; i < 4; i++)
+    {
+        if (cpsr->ge[i])
+            dest_word |= ((source1 >> (i * 8)) & 0xFF) << (i * 8);
```

Comment:
This would be clearer as `dest_word |= source1 & (0xFF << (i * 8));`. Same goes for the case below.